core/src/test/java/org/apache/calcite/sql/type/SqlTypeUtilTest.java

```diff
@@ -117,6 +120,47 @@
     SqlTypeCoercionRule.THREAD_PROVIDERS.set(defaultRules);
   }
 
+  @Test void testEqualAsCollectionSansNullability() {
+    final SqlTypeFactoryImpl factory =
+        new SqlTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
+    final RelDataType bigint = factory.createSqlType(SqlTypeName.BIGINT);
+
+    // case array
+    final RelDataType arrayNullable =
+        new ArraySqlType(factory.createTypeWithNullability(bigint, true), false);
+    final RelDataType arrayNotNull =
+        new ArraySqlType(factory.createTypeWithNullability(bigint, false), false);
+    assertThat(equalAsCollectionSansNullability(factory, arrayNullable, arrayNotNull),
+        is(true));
+
+    // case multiset
+    final RelDataType setNullable =
+        new MultisetSqlType(factory.createTypeWithNullability(bigint, true), false);
+    final RelDataType setNotNull =
+        new MultisetSqlType(factory.createTypeWithNullability(bigint, false), false);
+    assertThat(equalAsCollectionSansNullability(factory, setNullable, setNotNull),
+        is(true));
+
+    // multiset and array are not equal.
+    assertThat(equalAsCollectionSansNullability(factory, arrayNotNull, setNotNull),
+        is(false));
+  }
+
+  @Test void testEqualAsMapSansNullability() {
+    final SqlTypeFactoryImpl factory =
+        new SqlTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
+    final RelDataType bigint = factory.createSqlType(SqlTypeName.BIGINT);
```

Comment:
Use type from `SqlTypeFixture` if possible.

Reply:
> Use type from `SqlTypeFixture` if possible.

ok

Reply:
Thanks for your suggestion, updated already.